scipy/io/matlab/_miobase.py

```diff
@@ -421,7 +421,7 @@ def arr_to_chars(arr):
     arr = np.ndarray(shape=dims,
                      dtype=arr_dtype_number(arr, 1),
                      buffer=arr)
-    empties = [arr == '']
+    empties = [arr == np.array([''], dtype=arr_dtype_number(arr, 1))]
```

Comment:
```suggestion
    empties = [arr == np.array('', dtype=arr.dtype)]
```
Not sure if `arr` can be 0-D, but if it can be that would be wrong.

Reply:
Yeah I mean here I just fixed the issue.